src/go/rpk/pkg/cli/cmd/cluster/maintenance/enable.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package maintenance
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newEnableCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "enable <node-id>",
+		Short: "Enable maintenance mode for a node.",
+		Long: `Enable maintenance mode for a node.
+
+This command enables maintenance mode for the node with the specified ID. If a
+node exists that is already in maintenance mode then an error will be returned.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			node_id, err := strconv.Atoi(args[0])
+			if err != nil {
+				out.MaybeDie(err, "could not parse node id: %s: %v", args[0], err)
+			}
+
+			if node_id < 0 {
+				out.Die("invalid node id: %d", node_id)
+			}
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = client.EnableMaintenanceMode(node_id)
+			if he := (*admin.HttpError)(nil); errors.As(err, &he) {
+				if he.Response.StatusCode == 404 {
+					body, bodyErr := he.DecodeGenericErrorBody()
+					if bodyErr == nil {
```

Comment:
Do you mean `!=` ? Same for the similar line in the other branch below.

Reply:
Ah, it does catch the eye when reviewing Go, but this was intended.

The check above `if he := (*admin.HttpError)(nil); errors.As(err, &he) {` is meant to catch _any_ error. But then we want to get specific, and provide better error feedback. So if we observe that the error is a 404, then we want to decode the message further. In this case we assume that the response body for the 404 error contains a json body. So the line `body, bodyErr := he.DecodeGenericErrorBody()` still represents the 404 error. But, if we can successfully decode it (ie `bodyErr == nil` then we can provide a better error message `out.Die("Not found: %s", body.Message)`.